contracts/collections/reserves.ral

```diff
@@ -0,0 +1,89 @@
+Abstract Contract Reserves() {
+    const MaxAssetsStored = 8
+
+    @using(updateFields = true, preapprovedAssets = true)
+    fn handleReserves(caller: Address, subPath: ByteVec, tokenX: ByteVec, tokenY: ByteVec) -> (ByteVec, ByteVec) {
+        let deposit = mapEntryDeposit!()
+        let mut xId = #
+        let mut yId = #
+        let mut initX = false
+        let mut initY = false
+
+        if (reserves.contains!(tokenX)) {
+            xId = reserves[tokenX]
+        } else {
+            initX = true
+        }
+
+        if (reserves.contains!(tokenY)) {
+            yId = reserves[tokenY]
+        } else {
+            initY = true
+        }
+
+        if (initX && initY && lastReserveId == reserveTemplateId) {
+            let firstReserve = initReserve{caller -> ALPH: deposit}(caller, subPath, 2)
```

Comment:
can contract in `reserveTemplateId` be also used as first reserve? or does it need to stay empty?

Reply:
It needs to stay empty. We are passing invariant address as a parameter to the reserve, to deploy invariant the first reserve must be already deployed